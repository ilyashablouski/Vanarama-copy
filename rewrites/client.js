/* eslint-disable no-console */
const fetch = require('node-fetch');

const {
  ApolloClient,
  createHttpLink,
  InMemoryCache,
} = require('@apollo/client');

// const inspect = require('../inspect');

const httpLink = createHttpLink({
  uri: process.env.API_URL,
  fetch,
  headers: {
    'x-api-key': process.env.API_KEY,
  },
});

const client = new ApolloClient({
  link: httpLink,
  cache: new InMemoryCache(),
  onError: ({ networkError, graphQLErrors }) => {
    console.log('graphQLErrors', graphQLErrors);
    console.log('networkError', networkError);
  },
});

module.exports = client;
