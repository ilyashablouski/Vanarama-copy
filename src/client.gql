extend type Query {
  selectedOrderUuid: ID
  selectedDerivativeId: ID
}
