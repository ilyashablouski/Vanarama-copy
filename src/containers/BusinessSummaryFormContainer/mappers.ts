import { GetCreditApplicationByOrderUuid_creditApplicationByOrderUuid as CreditApplication } from '../../../generated/GetCreditApplicationByOrderUuid';
import { VehicleTypeEnum } from '../../../generated/globalTypes';

export const mapCreditApplicationToCreditChecker = (
  data?: CreditApplication | null,
  partyId?: string | null,
) => ({
  partyId: partyId || '',
  orderUuid: data?.lineItem?.order?.uuid,
  vehicleType:
    data?.lineItem?.vehicleProduct?.vehicleType || VehicleTypeEnum.CAR,
  capId: data?.lineItem?.vehicleProduct?.derivativeCapId,
  creditApplicationUuid: data?.lineItem?.creditApplications?.[0]?.uuid || '',
  monthlyPayment: data?.lineItem?.vehicleProduct?.monthlyPayment || 0,
  depositPayment: data?.lineItem?.vehicleProduct?.depositPayment || 0,
});

export default mapCreditApplicationToCreditChecker;
