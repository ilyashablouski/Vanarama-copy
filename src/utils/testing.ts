// eslint-disable-next-line import/no-extraneous-dependencies
import { fireEvent } from '@testing-library/react';

// eslint-disable-next-line import/prefer-default-export
export const inputChange = (element: Element, value: string) => {
  fireEvent.input(element, {
    target: { value },
  });
};
