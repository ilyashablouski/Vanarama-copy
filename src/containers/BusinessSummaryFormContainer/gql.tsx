import { gql } from '@apollo/client';
import BusinessSummaryForm from '../../components/BusinessSummaryForm/BusinessSummaryForm';
import AboutForm from '../../components/AboutForm';

export const GET_COMPANY_SUMMARY = gql`
  query GetCompanySummaryQuery($uuid: ID!, $personUuid: ID!) {
    companyByUuid(uuid: $uuid) {
      ...SummaryFormCompany
    }
    personByUuid(uuid: $personUuid) {
      ...AboutFormPerson
    }
  }
  ${BusinessSummaryForm.fragments.company}
  ${AboutForm.fragments.person}
`;

export default GET_COMPANY_SUMMARY;
