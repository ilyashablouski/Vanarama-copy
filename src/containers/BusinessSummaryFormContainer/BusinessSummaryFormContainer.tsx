import Loading from '@vanarama/uibook/lib/components/atoms/loading';
import React, { useEffect } from 'react';
import { useLazyQuery, ApolloError } from '@apollo/client';
import BusinessSummaryForm from '../../components/BusinessSummaryForm/BusinessSummaryForm';
import SoleTraderSummaryForm from '../../components/BusinessSummaryForm/SoleTraderSummaryForm';
import {
  GetCompanySummaryQuery,
  GetCompanySummaryQueryVariables,
  GetCompanySummaryQuery_personByUuid as PersonByUuid,
  GetCompanySummaryQuery_companyByUuid as CompanyByUuid,
} from '../../../generated/GetCompanySummaryQuery';
import { GET_COMPANY_SUMMARY, useUseFullCreditCheckerB2BMutation } from './gql';
import {
  useGetCreditApplicationByOrderUuid,
  useCreateUpdateCreditApplication,
} from '../../gql/creditApplication';
import { mapCreditApplicationToCreditChecker } from './mappers';
import { useGetPartyByUuidQuery } from '../../components/SummaryForm/gql';
import { GetPartyByUuid_partyByUuid as Party } from '../../../generated/GetPartyByUuid';
import { GetCreditApplicationByOrderUuid_creditApplicationByOrderUuid as CreditApplication } from '../../../generated/GetCreditApplicationByOrderUuid';

interface IProps {
  personUuid: string;
  companyUuid: string;
  orderId: string;
  isSoleTrader: boolean;
  onCompleted?: (emailAddresses: string | undefined) => void;
  onError?: (error: ApolloError) => void;
}

const BusinessSummaryFormContainer: React.FC<IProps> = ({
  companyUuid,
  orderId,
  personUuid,
  onCompleted,
  onError,
  isSoleTrader,
}) => {
  const [getDataSummary, getDataSummaryQueryOptions] = useLazyQuery<
    GetCompanySummaryQuery,
    GetCompanySummaryQueryVariables
  >(GET_COMPANY_SUMMARY, { fetchPolicy: 'no-cache' });
  const getCreditApplication = useGetCreditApplicationByOrderUuid(orderId);
  const [
    createUpdateCA,
    creditApplicationMutationOptions,
  ] = useCreateUpdateCreditApplication(orderId, () => {});
  const [
    submitFullCreditChecker,
    creditCheckerMutationOptions,
  ] = useUseFullCreditCheckerB2BMutation();
  const partyUuid =
    getCreditApplication.data?.creditApplicationByOrderUuid?.lineItem?.order
      ?.partyUuid;
  const getPartyByUuidQuery = useGetPartyByUuidQuery(partyUuid || '');

  const isSubmitting =
    creditApplicationMutationOptions.loading ||
    creditCheckerMutationOptions.loading;

  useEffect(() => {
    if (personUuid && companyUuid) {
      getDataSummary({
        variables: {
          uuid: companyUuid,
          personUuid,
        },
      });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [personUuid, companyUuid]);

  const error = getDataSummaryQueryOptions.error || getPartyByUuidQuery.error;
  if (error) {
    return <p>Error occurred: {error?.message}</p>;
  }

  if (
    getDataSummaryQueryOptions.loading ||
    getCreditApplication.loading ||
    getPartyByUuidQuery.loading ||
    (!getDataSummaryQueryOptions.data?.companyByUuid &&
      !getDataSummaryQueryOptions.data?.personByUuid)
  ) {
    return <Loading size="large" />;
  }

  const handleCreditApplicationSubmit = () =>
    createUpdateCA({
      variables: {
        input: {
          orderUuid: orderId,
          submittedAt: new Date(),
        },
      },
    });

  const handleCreditCheckerSubmit = (
    creditApplication?: CreditApplication | null,
    party?: Party | null,
  ) =>
    submitFullCreditChecker({
      variables: {
        input: mapCreditApplicationToCreditChecker(
          creditApplication,
          party?.company?.partyId || '',
        ),
      },
    });

  const handlePartyRefetch = (creditApplication?: CreditApplication | null) =>
    getPartyByUuidQuery.refetch({
      uuid: creditApplication?.lineItem?.order?.partyUuid || '',
    });

  const handleSubmit = () => {
    const personByUuid = getDataSummaryQueryOptions?.data?.personByUuid;
    handleCreditApplicationSubmit()
      .then(creditApplicationQuery =>
        handlePartyRefetch(
          creditApplicationQuery.data?.createUpdateCreditApplication,
        ).then(partyQuery =>
          handleCreditCheckerSubmit(
            creditApplicationQuery.data?.createUpdateCreditApplication,
            partyQuery.data?.partyByUuid,
          ),
        ),
      )
      .then(() => onCompleted?.(personByUuid?.emailAddresses[0].value))
      .catch(onError);
  };

  return (
    <>
      {isSoleTrader ? (
        <SoleTraderSummaryForm
          isSubmitting={isSubmitting}
          onSubmit={handleSubmit}
          creditApplication={
            getCreditApplication.data?.creditApplicationByOrderUuid
          }
          person={getDataSummaryQueryOptions.data.personByUuid as PersonByUuid}
          company={
            getDataSummaryQueryOptions.data.companyByUuid as CompanyByUuid
          }
        />
      ) : (
        <BusinessSummaryForm
          isSubmitting={isSubmitting}
          onSubmit={handleSubmit}
          creditApplication={
            getCreditApplication.data?.creditApplicationByOrderUuid
          }
          person={getDataSummaryQueryOptions.data.personByUuid as PersonByUuid}
          company={
            getDataSummaryQueryOptions.data.companyByUuid as CompanyByUuid
          }
        />
      )}
    </>
  );
};

export default BusinessSummaryFormContainer;
