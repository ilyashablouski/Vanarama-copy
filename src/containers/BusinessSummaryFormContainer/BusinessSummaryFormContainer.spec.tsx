import { MockedProvider } from '@apollo/client/testing'; 
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import BusinessSummaryFormContainer from './BusinessSummaryFormContainer';
import createCompanyData from './__fixtures__/testCompanyData';
import g from '../../components/BusinessSummaryForm/BusinessSummaryFormDirectorDetailsSection'

const mockPush = jest.fn();
jest.mock('next/router', () => ({
  useRouter() {
    return {
      prefetch: () => null,
      push: mockPush,
    };
  },
}));

describe('<BusinessSummaryFormContainer />', () => {
  it('should render company details correctly', async () => {
    // ARRANGE
    const uuid = 'ad0f772b-eded-483a-96be-18ea4e67948d';
    const mocks = [createCompanyData(uuid)];

    // ACT
    render(
      <MockedProvider addTypename={false} mocks={mocks}>
        <BusinessSummaryFormContainer companyUuid={uuid} />
      </MockedProvider>,
    );

    // Wait for the data to load
    await screen.findByText('Company Details');

    // ASSERT
    expect(screen.getByTestId(/summary-business-name/)).toHaveTextContent(
      'Nastia Test2',
    );

    expect(screen.getByTestId(/summary-company-number/)).toHaveTextContent(
      '09876546',
    );

    expect(screen.getByTestId(/summary-business-nature/)).toHaveTextContent(
      'Fairy tale',
    );

    expect(screen.getByTestId(/summary-registered-address/)).toHaveTextContent(
      'Desafinado, 15 Mill Bridge Close, Crewe, CW1 5DZ'
      );
    expect(screen.getByTestId(/summary-trading-address/)).toHaveTextContent(
      'Sadlers Farm, Lower Pennington Lane, Lymington, SO41 8AL',
    );

    expect(screen.getByTestId(/summary-trading-since/)).toHaveTextContent(
      'February 2005',
    );

    expect(screen.getByTestId(/summary-telephone-number/)).toHaveTextContent(
      '09876 543211',
    );

    expect(screen.getByTestId(/summary-email-address/)).toHaveTextContent('a.harbuz81@reply.com');
  });

  it('should render VAT details correctly', async () => {
    // ARRANGE
    const uuid = 'ad0f772b-eded-483a-96be-18ea4e67948d';
    const mocks = [createCompanyData(uuid)];

    // ACT
    render(
      <MockedProvider addTypename={false} mocks={mocks}>
        <BusinessSummaryFormContainer companyUuid={uuid} />
      </MockedProvider>,
    );

    // Wait for the data to load
    await screen.findByText('VAT Details');

    // ASSERT
    expect(screen.getByTestId('summary-vat-details')).toHaveTextContent(
      '123456789',
    );

    expect(screen.getByTestId('summary-vat-countries')).toHaveTextContent(
      'Belarus Vanuatu',
    );

    expect(screen.getByTestId('summary-turnover-percentage')).toHaveTextContent(
      '40%',
    );
  });

  it('should render company bank details correctly', async () => {
    // ARRANGE
    const uuid = 'ad0f772b-eded-483a-96be-18ea4e67948d';
    const mocks = [createCompanyData(uuid)];

    // ACT
    render(
      <MockedProvider addTypename={false} mocks={mocks}>
        <BusinessSummaryFormContainer companyUuid={uuid} />
      </MockedProvider>,
    );

    // Wait for the data to load
    await screen.findByText('Company Bank Details');

    // ASSERT
    expect(screen.getByTestId('summary-name-on-card')).toHaveTextContent(
      'Nastia',
    );

    expect(screen.getByTestId('summary-sort-code')).toHaveTextContent(
      '11-22-33',
    );

    expect(screen.getByTestId('summary-account-number')).toHaveTextContent(
      '12345678',
    );
  });

  it('should render directors correctly and sorted by sharehold percentage desc', async () => {
    // ARRANGE
    const uuid = 'ad0f772b-eded-483a-96be-18ea4e67948d';
    const mocks = [createCompanyData(uuid)];

    console.log('/////////////////////');
    console.log((mocks[0].result as any).data.companyByUuid.associates);

    // ACT
    render(
      <MockedProvider addTypename={false} mocks={mocks}>
        <BusinessSummaryFormContainer companyUuid={uuid} />
      </MockedProvider>,
    );

    // Wait for the data to load
    await waitFor(() => {
      expect(screen.getByText('Female')).toBeInTheDocument();
    });
    // await screen.findByText('Director Details');
    // console.log('//////////////////////////////////////');
    // console.log(screen.getByTestId('company_director_details_heading_[0]'));

    // ASSERT
    expect(
      screen.getByTestId('company_director_details_heading_[0]'),
    ).toHaveTextContent('Anastasiya Harbuz');

    expect(screen.getByTestId('summary-director[0]-title')).toHaveTextContent(
      'Miss',
    );

    expect(
      screen.getByTestId('summary-director[0]-gender'),
    ).toHaveTextContent('Female');

    expect(screen.getByTestId('summary-director[0]-birth-date')).toHaveTextContent(
      '17 February 1999',
    );

    expect(
      screen.getByTestId('summary-director[0]-business-share'),
    ).toHaveTextContent('40%');

    expect(screen.getByTestId('summary-director[0]-noOfDependants')).toHaveTextContent(
      'None',
    );

    expect(
      screen.getByTestId('summary-director[0]-curr-address'),
    ).toHaveTextContent('Sky View, 5 Harrogate Road, York, YO51 9JD');

    expect(
      screen.getByTestId('summary-director[0]-curr-moved-in'),
    ).toHaveTextContent('February 2007');

    expect(screen.getByTestId('summary-director[0]-curr-prop-status')).toHaveTextContent(
      'Owned with mortgage',
    );
  });

  it('should redirect to the thank you page when clicking "Continue"', async () => {
    // ARRANGE
    const uuid = 'ad0f772b-eded-483a-96be-18ea4e67948d';
    const mocks = [createCompanyData(uuid)];

    // ACT
    render(
      <MockedProvider addTypename={false} mocks={mocks}>
        <BusinessSummaryFormContainer companyUuid={uuid} />
      </MockedProvider>,
    );

    // Wait for the data to load
    await screen.findByTestId('summary-heading');

    fireEvent.click(screen.getByText(/Continue/));

    // ASSERT
    expect(mockPush).toHaveBeenCalledTimes(1);
    expect(mockPush).toHaveBeenCalledWith('/olaf/thank-you');
  });

  it('should redirect to about page when clicking "Edit" on the "Your Details" section', async () => {
    // ARRANGE
    const uuid = 'fd2333b8-6da1-47d2-837d-bc69849e0764';
    const mocks = [createBruceData(uuid)];

    // ACT
    render(
      <MockedProvider addTypename={false} mocks={mocks}>
        <BusinessSummaryFormContainer companyUuid={uuid} />
      </MockedProvider>,
    );

    // Wait for the data to load
    await screen.findByTestId('summary-heading');

    fireEvent.click(screen.getByTestId(/edit-your-details/));

    // ASSERT
    expect(mockPush).toHaveBeenCalledTimes(1);
    expect(mockPush).toHaveBeenCalledWith(
      '/olaf/about/[uuid]?redirect=summary',
      '/olaf/about/fd2333b8-6da1-47d2-837d-bc69849e0764?redirect=summary',
    );
  });

  it('should redirect to employment history page when clicking "Edit" on the "Employment History" section', async () => {
    // ARRANGE
    const uuid = 'fd2333b8-6da1-47d2-837d-bc69849e0764';
    const mocks = [createBruceData(uuid)];

    // ACT
    render(
      <MockedProvider addTypename={false} mocks={mocks}>
        <BusinessSummaryFormContainer companyUuid={uuid} />
      </MockedProvider>,
    );

    // Wait for the data to load
    await screen.findByTestId('summary-heading');

    fireEvent.click(screen.getByTestId(/edit-employment-history/));

    // ASSERT
    expect(mockPush).toHaveBeenCalledTimes(1);
    expect(mockPush).toHaveBeenCalledWith(
      '/olaf/employment-history/[uuid]?redirect=summary',
      '/olaf/employment-history/fd2333b8-6da1-47d2-837d-bc69849e0764?redirect=summary',
    );
  });

  it('should redirect to expenses page when clicking "Edit" on the "Monthly Income" section', async () => {
    // ARRANGE
    const uuid = 'fd2333b8-6da1-47d2-837d-bc69849e0764';
    const mocks = [createBruceData(uuid)];

    // ACT
    render(
      <MockedProvider addTypename={false} mocks={mocks}>
        <BusinessSummaryFormContainer companyUuid={uuid} />
      </MockedProvider>,
    );

    // Wait for the data to load
    await screen.findByTestId('summary-heading');

    fireEvent.click(screen.getByTestId(/edit-expenses/));

    // ASSERT
    expect(mockPush).toHaveBeenCalledTimes(1);
    expect(mockPush).toHaveBeenCalledWith(
      '/olaf/expenses/[uuid]?redirect=summary',
      '/olaf/expenses/fd2333b8-6da1-47d2-837d-bc69849e0764?redirect=summary',
    );
  });

  it('should redirect to bank details page when clicking "Edit" on the "Bank Details" section', async () => {
    // ARRANGE
    const uuid = 'fd2333b8-6da1-47d2-837d-bc69849e0764';
    const mocks = [createBruceData(uuid)];

    // ACT
    render(
      <MockedProvider addTypename={false} mocks={mocks}>
        <BusinessSummaryFormContainer companyUuid={uuid} />
      </MockedProvider>,
    );

    // Wait for the data to load
    await screen.findByTestId('summary-heading');

    fireEvent.click(screen.getByTestId(/edit-bank-details/));

    // ASSERT
    expect(mockPush).toHaveBeenCalledTimes(1);
    expect(mockPush).toHaveBeenCalledWith(
      '/olaf/bank-details/[uuid]?redirect=summary',
      '/olaf/bank-details/fd2333b8-6da1-47d2-837d-bc69849e0764?redirect=summary',
    );
  });
});
