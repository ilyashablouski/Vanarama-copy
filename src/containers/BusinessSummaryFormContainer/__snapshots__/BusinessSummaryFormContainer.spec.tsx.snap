// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessSummaryFormContainer /> should render company bank details correctly 1`] = `
<body>
  <div>
    <form
      class="form olaf--summary"
    >
      <p
        class="heading -black -xlarge"
        data-testid="summary-heading"
      >
        Summary
      </p>
      <section
        class="structured-list"
      >
        <div
          class="structured-list-thead"
        >
          <div
            class="structured-list-row structured-list-row--header-row"
          >
            <div
              class="structured-list-th"
            >
              <p
                class="text -large -darker"
                data-testid="company_details_heading_data_testId"
              >
                Company Details
              </p>
            </div>
          </div>
        </div>
        <div
          class="structured-list-tbody"
        >
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-0-Business Name"
            >
              Business Name
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-business-name"
            >
              Nastia Test2
            </div>
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-1-Business Registration Number"
            >
              Business Registration Number
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-company-number"
            >
              09876546
            </div>
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-2-Nature of Business"
            >
              Nature of Business
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-business-nature"
            >
              Fairy tale
            </div>
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-3-Registered Address"
            >
              Registered Address
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-registered-address"
            >
              Desafinado, 15 Mill Bridge Close, Crewe, CW1 5DZ
            </div>
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-4-Trading Address"
            >
              Trading Address
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-trading-address"
            >
              Sadlers Farm, Lower Pennington Lane, Lymington, SO41 8AL
            </div>
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-5-Trading Since"
            >
              Trading Since
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-trading-since"
            >
              February 2005
            </div>
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-6-Business Telephone Number"
            >
              Business Telephone Number
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-telephone-number"
            >
              09876 543211
            </div>
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-7-Email"
            >
              Email
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-email-address"
            >
              a.harbuz81@reply.com
            </div>
          </div>
        </div>
        <div
          class="structured-list--edit"
        >
          <a
            class="link -teal -regular -plain"
            data-testid="edit-company-details"
          >
            Edit
          </a>
        </div>
      </section>
      <br />
      <section
        class="structured-list"
      >
        <div
          class="structured-list-thead"
        >
          <div
            class="structured-list-row structured-list-row--header-row"
          >
            <div
              class="structured-list-th"
            >
              <p
                class="text -large -darker"
                data-testid="company_vat_details_heading_data_testId"
              >
                VAT Details
              </p>
            </div>
          </div>
        </div>
        <div
          class="structured-list-tbody"
        >
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-0-VAT Number"
            >
              VAT Number
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-vat-details"
            >
              123456789
            </div>
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-1-Countries You Trade In"
            >
              Countries You Trade In
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-vat-countries"
            >
              

Belarus

Vanuatu
            </div>
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-2-Percentage Of Company Turnover Outside The UK"
            >
              Percentage Of Company Turnover Outside The UK
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-turnover-percentage"
            >
              40%
            </div>
          </div>
        </div>
        <div
          class="structured-list--edit"
        >
          <a
            class="link -teal -regular -plain"
            data-testid="edit-vat-details"
          >
            Edit
          </a>
        </div>
      </section>
      <section
        class="structured-list olaf--summary-title"
      >
        <div
          class="structured-list-thead"
        >
          <div
            class="structured-list-row structured-list-row--header-row"
          >
            <div
              class="structured-list-th"
            >
              <p
                class="text -large -darker"
                data-testid="bank_details_heading_data_testId"
              >
                Company Bank Details
              </p>
            </div>
          </div>
        </div>
        <div
          class="structured-list-tbody"
        >
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-0-Name on Card"
            >
              Name on Card
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-name-on-card"
            >
              Nastia
            </div>
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-1-Sort Code"
            >
              Sort Code
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-sort-code"
            >
              11-22-33
            </div>
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-2-Account Number"
            >
              Account Number
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-account-number"
            >
              12345678
            </div>
          </div>
        </div>
        <div
          class="structured-list--edit"
        >
          <a
            class="link -teal -regular -plain"
            data-testid="edit-bank-details"
          >
            Edit
          </a>
        </div>
      </section>
      <br />
      <p
        class="heading olaf--summary-title -black -large"
        data-testid="directors-section-heading"
      >
        Director Details
      </p>
      <hr />
      <section
        class="structured-list -styled-headers"
      >
        <div
          class="structured-list-thead"
        >
          <div
            class="structured-list-row structured-list-row--header-row"
          >
            <div
              class="structured-list-th"
            >
              <p
                class="text -lead -darker"
                data-testid="company_director_details_heading_[0]"
              >
                undefined undefined
              </p>
            </div>
          </div>
        </div>
        <div
          class="structured-list-tbody"
        >
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-0-Title"
            >
              Title
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director[0]-title"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-1-Gender"
            >
              Gender
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director[0]-gender"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-2-Date Of Birth"
            >
              Date Of Birth
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director[0]-birth-date"
            >
              20 July 2020
            </div>
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-3-% Shareholder Of Business"
            >
              % Shareholder Of Business
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director[0]-business-share"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-4-Number of Dependants"
            >
              Number of Dependants
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director[0]-noOfDependants"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-5-Current Address"
            >
              Current Address
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director[0]-curr-address"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-6-Date Moved In"
            >
              Date Moved In
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director[0]-curr-moved-in"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-7-Property Status"
            >
              Property Status
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director-curr-prop-status[0]"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-8-Past Address"
            >
              Past Address
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director-past-address[0]"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-9-Date Moved In"
            >
              Date Moved In
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director-past-moved-in[0]"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-10-Property Status"
            >
              Property Status
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director-past-prop-status[0]"
            />
          </div>
        </div>
        <div
          class="structured-list--edit"
        >
          <a
            class="link -teal -regular -plain"
            data-testid="edit-director-details[0]"
          >
            Edit
          </a>
        </div>
      </section>
      <section
        class="structured-list -styled-headers"
      >
        <div
          class="structured-list-thead"
        >
          <div
            class="structured-list-row structured-list-row--header-row"
          >
            <div
              class="structured-list-th"
            >
              <p
                class="text -lead -darker"
                data-testid="company_director_details_heading_[1]"
              >
                undefined undefined
              </p>
            </div>
          </div>
        </div>
        <div
          class="structured-list-tbody"
        >
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-0-Title"
            >
              Title
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director[1]-title"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-1-Gender"
            >
              Gender
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director[1]-gender"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-2-Date Of Birth"
            >
              Date Of Birth
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director[1]-birth-date"
            >
              20 July 2020
            </div>
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-3-% Shareholder Of Business"
            >
              % Shareholder Of Business
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director[1]-business-share"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-4-Number of Dependants"
            >
              Number of Dependants
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director[1]-noOfDependants"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-5-Current Address"
            >
              Current Address
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director[1]-curr-address"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-6-Date Moved In"
            >
              Date Moved In
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director[1]-curr-moved-in"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-7-Property Status"
            >
              Property Status
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director-curr-prop-status[1]"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-8-Past Address"
            >
              Past Address
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director-past-address[1]"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-9-Date Moved In"
            >
              Date Moved In
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director-past-moved-in[1]"
            />
          </div>
          <div
            class="structured-list-row "
          >
            <div
              class="structured-list-td structured-list-content--nowrap"
              data-testid="data_id-10-Property Status"
            >
              Property Status
            </div>
            <div
              class="structured-list-td"
              data-testid="summary-director-past-prop-status[1]"
            />
          </div>
        </div>
        <div
          class="structured-list--edit"
        >
          <a
            class="link -teal -regular -plain"
            data-testid="edit-director-details[1]"
          >
            Edit
          </a>
        </div>
      </section>
      <button
        class="button -mt-400 -teal -large -solid"
        data-testid="olaf_summary_continue_buttton"
        type="button"
      >
        <div
          class="button--inner"
        >
          Continue
        </div>
      </button>
    </form>
  </div>
</body>
`;
